types/bytebuffer/index.d.ts

```diff
@@ -316,7 +317,8 @@ declare class ByteBuffer
     /**
      * Reads a NULL-terminated UTF8 encoded string. For this to work the string read must not contain any NULL characters itself.
      */
-    readCString( offset?: number ): string;
+    readCString(): string;
+    readCString( offset: number ): { string: string, length: number};
```

Comment:
I cannot agree.
Please take a look into docs.
` { string: string, length: number}` is returned when and only when offset is provided.
When there is no offset provided, then just `string` is returned.

My change make typing exactly as they are in the runtime.
If it brakes something to someone, it means it was broken before.
Having `string | { string: string, length: number}` is bad practice, because you then have to cast later on to expected result and that is exactly why I came here to fix it because we have plenty of castings in our codebase around this functions.

Reply:
OK, clear. I wrote the definitions years ago and I need some info to understand. What about these methods:

 writeVString( str: string, offset?: number ): ByteBuffer | number;
    /**
     * Writes a 32bit base 128 variable-length integer.
     */
    writeVarint32( value: number, offset?: number ): ByteBuffer | number;
    /**
     * Writes a zig-zag encoded 32bit base 128 variable-length integer.
     */
    writeVarint32ZigZag( value: number, offset?: number ): ByteBuffer | number;
    /**
     * Writes a 64bit base 128 variable-length integer.
     */
    writeVarint64( value: number | Long, offset?: number ): ByteBuffer;
    /**
     * Writes a zig-zag encoded 64bit base 128 variable-length integer.
     */
    writeVarint64ZigZag( value: number | Long, offset?: number ): ByteBuffer | number;

Do we need to split the returning value based on signature or are they just fine as they are?

Reply:
Cool.
I thought I've found all of these issues but looks like there is more.
I will do that either in this PR or next one.
It's a shame there is this stupid issue with your username and your approval is not treated as maintainer's :/

Reply:
Please fix those methods in this PR and then I’ll approve.

Reply:
Done! Please take a look and approve.